snake_n_ladder.rb

```diff
@@ -0,0 +1,80 @@
+def snake_ladder (position, dice, snake_hash, snake_message, ladders_hash, ladder_message)
+    if position + dice >100
+        puts "You just need a #{100-position}"
+        return position
+    end
+    position = position + dice
+    snake_pos = snake_hash[position]
+    ladder_pos = ladders_hash[position]
+    if snake_pos.class != NilClass
+        puts snake_message
+        puts "Tooks you down from #{position} to #{snake_pos}"
+        position = snake_pos
+    elsif ladder_pos.class != NilClass
+        puts ladder_message
+        puts "Tooks you up from #{position} to #{ladder_pos}"
+        position = ladder_pos
+    end
+    position
+end
+#Start messages for the game
+puts "\t\tSNAKE AND LADDERS GAME"
+puts "Enter your name"
+puts "Player 1:"
+player1 = gets.chomp
+if player1 == ""
```

Comment:
no need to check condition just initialise `player1` and `player2` with the default names